src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          if (key in state) {
+            delete state[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
```

Comment:
You should add a default case to the switch/case constructions. For example
```suggestion
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;
        
        default: 
        break;
```